### PR TITLE
Update documentation for improvement in rasa sdk knowledge base actions based on improvements suggested in PR#922 in rasa-sdk

### PR DESCRIPTION
**Pull request by Gajithra**
**Proposed changes**:
- Updated Knowledge Base Actions doc for the improvement related to the PR `OSS-668: Rasa Knowledge base actions are unable to query about a certain attribute of an object unless the user first asks to obtain a list of objects of a specific type`. Link to this PR is accessible [here](https://github.com/RasaHQ/rasa-sdk/pull/922). 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Tawakalt) [inline docs/docs/action-server/knowledge-base-actions.mdx:314]:
```suggestion
attribute of interest. The user is not required to query the knowledge base to list any kind of object prior to this.
```

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Tawakalt) [inline docs/docs/action-server/knowledge-base-actions.mdx:283]:
```suggestion
In order to query the knowledge base to list any kind of object, the user's request needs to include the object type.
```

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Tawakalt) [inline docs/docs/action-server/knowledge-base-actions.mdx:315]:
```suggestion
The `ActionQueryKnowledgeBase` will extract the object type from the user's request and query the knowledge base for an attribute of the object. 
```

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Tawakalt) [inline docs/docs/action-server/knowledge-base-actions.mdx:334]:
```suggestion
to extract the object type of the object the user is interested in.
```

**Review by Tawakalt — COMMENTED** [review]
Thanks for your contribution. 💯 
I left some few suggestions. 

We can merge this only after the SDK PR has been merged.

**Review by Tawakalt — COMMENTED** [review]
We also need a changelog of type `doc`

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Gajithra) [reply docs/docs/action-server/knowledge-base-actions.mdx]: Done ✅

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Gajithra) [reply docs/docs/action-server/knowledge-base-actions.mdx]: Done ✅

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Gajithra) [reply docs/docs/action-server/knowledge-base-actions.mdx]: Done :+

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Gajithra) [reply docs/docs/action-server/knowledge-base-actions.mdx]: Done 👍🏽

**Comment by Gajithra** [comment]
> 
Noted and Thanks for your feedback @Tawakalt ! I have made the requested changes.

File: changelog/12362.doc.md
Comment (ancalita) [reply changelog/12362.doc.md:1]: I agree, this should be a brief note summarising the docs updates.

**Review by ancalita — COMMENTED** [review]
Looks good ⚡ I left some further suggestions below.

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Gajithra) [reply docs/docs/action-server/knowledge-base-actions.mdx]: Done 👍🏽

File: docs/docs/action-server/knowledge-base-actions.mdx
Comment (Gajithra) [reply docs/docs/action-server/knowledge-base-actions.mdx]: Got it. I incorporated the changes Anca Lita

File: changelog/12362.doc.md
Comment (Gajithra) [reply changelog/12362.doc.md]: Thanks for the suggestion @Tawakalt and @ancalita. I have updated the changelog.

**Review by ancalita — APPROVED** [review]
🎉 
As soon as you update your branch with latest `main`, we can merge this as well.

**Comment by Gajithra** [comment]
> 🎉 As soon as you update your branch with latest `main`, we can merge this as well.

Awesome @ancalita ! We have updated the branch now

**Pull request merged**
